components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/PlacesHistoryStorage.kt

```diff
@@ -293,4 +293,25 @@ open class PlacesHistoryStorage(
             }
         }
     }
+
+    override suspend fun deleteHistoryMetadata(searchTerm: String) {
+        // Ideally, we want this to live in A-S as a simple DELETE statement.
+        // As-is, this isn't an atomic operation. For how we're using these data, both lack of
+        // atomicity and a performance penalty is acceptable for now.
+        withContext(writeScope.coroutineContext) {
+            handlePlacesExceptions("deleteHistoryMetadataSearchGroup") {
+                places.reader().getHistoryMetadataSince(Long.MIN_VALUE)
+                    .filter { it.searchTerm == searchTerm }
```

Comment:
One thing we have to be careful about here when integrating is that this is case sensitive. Looks like in the UI we always show lower case even if it's not. Should we just compare lowercase strings here?